File: ice0208/linked-list/206.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def reverseList(self, head):
        if head == None or head.next == None:
            return head

        prev = head
        cur = head.next
        head.next = None
        while cur:
            next = cur.next
            cur.next = prev
            prev = cur
            cur = next

        return prev

File: moonpoison/linked_list/206.cpp

#include<iostream>
using namespace std;
struct ListNode 
{
    int val;
    ListNode *next;
    ListNode() : val(0), next(nullptr) {}
    ListNode(int x) : val(x), next(nullptr) {}
    ListNode(int x, ListNode *next) : val(x), next(next) {}
};
class Solution {
public:
    ListNode* reverseList(ListNode* head) 
    {
        ListNode *first=head, *second=NULL;
        if(head==NULL) return head;
        for(ListNode *tmp=first->next;tmp!=NULL;tmp=tmp->next)
        {
            first->next=second;
            second=first;
            first=tmp;
        }
        first->next=second;
        return first;
    }
};